src/components/Admin/ListDevices/index.js

```diff
@@ -0,0 +1,149 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import PropTypes from 'prop-types';
+import uiActions from '../../../redux/actions/ui';
+import MaterialTable from 'material-table';
+import { fetchDevices } from '../../../apis/index';
+import { DEVICE } from './constants';
+import { ActionDiv } from '../../shared/TableActionStyles';
+import { removeUserDevice } from '../../../apis/index';
+
+class ListDevices extends React.Component {
+  state = {
+    devices: [],
+    loadingDevices: true,
+    macId: '',
+    email: '',
+  };
+
+  componentDidMount() {
+    this.loadDevices();
+  }
+
+  loadDevices = () => {
+    fetchDevices()
+      .then(payload => {
+        const { devices } = payload;
+        let devicesArray = [];
+        devices.forEach(device => {
+          const email = device.name;
+          const devices = device.devices;
+          const macIdArray = Object.keys(devices);
+          macIdArray.forEach(macId => {
+            const device = devices[macId];
+            let location = 'Location not given';
+            if (device.geolocation) {
+              location = `${device.geolocation.latitude},${device.geolocation.longitude}`;
+            }
+            let macid = macId
```

Comment:
Why have you used this manipulation? Simply pass  `shouldCameliseKeys : false` in the API definition